src/vmm/src/devices/virtio/queue.rs

```diff
@@ -1073,10 +1071,52 @@ mod verification {
 
     #[kani::proof]
     #[kani::unwind(0)]
-    fn verify_set_used_ring_avail_event() {
+    fn verify_avail_ring_idx_get() {
+        let ProofContext(queue, _) = ProofContext::bounded_queue();
```

Comment:
If I replace this one with `kani::any()`, I get a failure about a null pointer dereference. It's because `impl kani::Arbitrary for ProofContext` is missing `queue.set_pointers`. If I add that, then I can apply the following diff

```suggestion
        let ProofContext(queue, _) = kani::any();
```

which passes in 25 seconds, and is a waaaay more general proof

Reply:
Thanks! Could you also do it for the pre-existing ones?